CMakeLists.txt

```diff
@@ -24,6 +24,8 @@ otherwise a crypto target needs to be defined." ON)
 option(UNSAFE_TREAT_WARNINGS_AS_ERRORS "Compiler warnings are treated as errors. Warnings may
 indicate danger points where you should verify with the S2N-TLS developers that the security of
 the library is not compromised. Turn this OFF to ignore warnings." ON)
+option(S2N_WERROR_ALL "This option will cause all artifacts linked to libs2n to use the 
```

Comment:
Should this be set in s2n_codebuild.sh for unit tests?

Reply:
Yes, but that is a much larger project unfortunately. That was my initial approach, but I encountered some of the following problems
- 32 bit build has a rollover warning in the 2050 detection function (which will never be called, but the time_t definition has that problem regardless
- individual librypto builds have problems with unused variable/unused function warnings, because there are some functions that never get called under certain if/def conditions.